tests/all/integration/api/helpers/order/test_validate_discount_code.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime
+
+import pytest
+
+from app.api.helpers.exceptions import UnprocessableEntity
+from app.api.helpers.ticketing import validate_discount_code
+from tests.factories.attendee import AttendeeSubFactory
+from tests.factories.discount_code import DiscountCodeTicketSubFactory
+from tests.factories.event import EventFactoryBasic
+from tests.factories.order import OrderSubFactory
+from tests.factories.ticket import TicketSubFactory
+
+
+def _create_discount_code(db, **kwargs):
+    tickets = TicketSubFactory.create_batch(3, event=EventFactoryBasic(), **kwargs)
+    discount = DiscountCodeTicketSubFactory(tickets_number=5, tickets=tickets)
+    db.session.commit()
+
+    return discount, [{'id': ticket.id} for ticket in tickets]
+
+
+def test_validate_discount_code(db):
+    discount, tickets = _create_discount_code(db)
+
+    assert discount.validate(tickets=tickets) == discount
+
+
+def test_validate_discount_code_id(db):
+    discount, tickets = _create_discount_code(db)
+
+    assert validate_discount_code(discount.id, tickets=tickets) == discount
+    assert validate_discount_code(str(discount.id), tickets=tickets) == discount
+
+
+def test_validate_discount_code_require_tickets_or_holders(db):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function name "test_validate_discount_code_require_tickets_or_holders" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5524612)